### ISSUE TITLE
[Feature Request] Allow requiring user to fill-up custom environment variables before install

### ISSUE THREAD

**Issue by WisdomSky** [IceWhaleTech/CasaOS#1248]
**Is your feature request related to a problem? Please describe.**

Unable to adapt certain docker images as a CasaOS app. 

There are many docker images out there that requires the user to specify a certain information like API_KEYs, usernames and passwords, etc into the[ `environment`](https://docs.docker.com/compose/environment-variables/set-environment-variables/#use-the-environment-attribute) property or via a [`.env`](https://docs.docker.com/compose/environment-variables/set-environment-variables/#use-the-environment-attribute) file if using a docker-compose. 

The current implementation in CasaOS is that everything must be hard-coded into the `docker-compose.yml` and there's no way for the user to provide these required information beforehand. 

Stopping the container and opening the settings and setting the environment variable will not work for all images as some images initialize only once (which is at the time the container is created). Stopping/Starting or Restarting the container will not re-initialize the container.

**Describe the solution you'd like**

##### Specify Required Environment Variables

Give app developers the ability to specify in the docker-compose file environment variables for each service that the user is required to fill-up before the container is created.

**YAML pseudo-code:**
```yaml
name: something
services:
  someapp:
    image: someuser/someimage:1.0
    environment:
      SOME_API_KEY: ''
    depends_on:
      - somedb
    ...
  somedb:
    image: postgres:13
    environment:
      DB_USER: 'user'
      DB_PASSWORD: 'pass'
    ...
x-casaos:
  main: someapp
  ...
  before_install:
    environment:
      someapp:
        - name: SOME_API_KEY
          description:
            en_us: "Some API Key"
      somedb:
        - name: DB_USER
          description:
            en_us: "Preferred database username"
        - name: DB_PASSWORD
          description:
            en_us: "Preferred database password"
  ...
  port_map: "8008"
```

**Describe alternatives you've considered**
No alternative applicable.

**Additional context**
No additional context.

**Comment by tigerinus** [comment]
sounds like a duplicate of

- #1101

Closing this one - Let's track this feature request in #1101.

**Comment by tigerinus** [comment]
Additionally, @CorrectRoadH has added a feature (CLI only for now) in v0.4.4 to set global environment variables which will be passed to each app later on.

E.g. you can set `OPENAI_API_KEY` globally, so every app depends on OpenAI no longer needs this env var set separately.

Check the CLI command out and see if it addresses your need somehow.
